src/Types.jl

```diff
@@ -630,7 +635,9 @@ function write_env_usage(source_file::AbstractString, usage_filepath::AbstractSt
         end
 
         # record new usage
-        usage[source_file] = [Dict("time" => timestamp)]
+        for source_file in source_files
+            usage[source_file] = [Dict("time" => timestamp)]
```

Comment:
As this has been motivated by hunting down unnecessary allocations: Do these have to be seperate objects per `source_file`? (both the single-entry vector and the dict)

Reply:
Don't think they have to be separate but the cost savings will be extremely low, (7.38MB -> 7.34MB from a quick test). Could maybe do it as an aesthetic thing.

Reply:
Ok, nevermind then. Just wanted to be sure it's a conscious decision, not an oversight.